### ISSUE TITLE
Quick help on vulcanisation

### ISSUE THREAD

**Issue by mercmobily** [PolymerElements/polymer-starter-kit#936]
I realise this is not strictly a PSK question. But, I don't know where to turn...
I am speed-developing an app. It's online, although it's not even 10% finished: http://www.entertainers.biz/ Load time is horrific. 7 to 13 seconds... and the app isn't even there yet!
I changed the pattern slightly from the PSK:

* index.html loads `<my-main>`
* `<my-main>` displays a loader while it makes an AJAX call to figure out things about the user (logged in? Admin? etc.). So, `<my-main>` needs a bunch of elements from polymer. for the record, `<my-main>` also imports `<my-app>`.
* `<my-main>` will display `my-app` (once the AJAX call is done), which in turns loads another bunch of modules.
* `<my-app>` is the typical PSK app with a drawer, etc. and doing lazy loading of sub-pages
* Sub-pages themselves might import more modules

With `polymer build` I save a couple of seconds. However, 230 HTTP requests still take time. Also, I am not 100% clear what `polymer build` *actually* do, and cannot *quite* decide if the "shell" of my app is `my-main.html` (the pre-screen that loads the app using AJAX) or `my-app.html`. I also don't quite understand how `polymer serve` actually *uses* the information... the entry point, yes, I understand. But the "shell"...?

Now... if I were to decide that I want to use vulcanize to reduce the calls *dramatically*, how would I go? (Disclaimer: my understanding of vulcanize is that it will create a single file replacing "in line" the imported HTML files)

This is my current idea: create a modules that imports ALL of the modules needed by `my-main` and `my-app` (call it `my-core-libs`), and load *that* one from `my-main` and `my-app`. ALSO, import `core-libs` whenever a file needs any ONE of the bundled files

Is this the way to go? And am I right in saying that I need to be SUPER careful NOT to import, from another file, a module that was ALREADY imported in `my-core-libs`? (otherwise kaboom, Polymer tries to register the same element twice etc.)

Any help, ideas, enlightenment, would be *amazingly* welcome (assuming you guys have any time and inclination to get to ticket like this one after the super Polymer summit!!!)

THANKS!

**Comment by mercmobily** [comment]
and in fact, how would you vulcanise the PSK itself?
For example `my-app.html` has:

    https://github.com/PolymerElements/polymer-starter-kit/blob/master/src/my-app.html
    <link rel="import" href="../bower_components/polymer/polymer.html">
    <link rel="import" href="shared-styles.html">

And `my-view1.html` has:

    <link rel="import" href="../bower_components/polymer/polymer.html">
    <link rel="import" href="shared-styles.html">

So if I vulcanise `my-app.html`, I would then need to make sure that `my-view1.html` DOESN'T load `../bower_components/polymer/polymer.html` again...?

Or am I talking total nonsense?

**Comment by mercmobily** [comment]
I am also sooooooo confused as in what `polymer build` does! The docs talk about vulcanization, but `vulcanize.ts` was ripped out and there are NO results if you look for the word `vulcanize` in the source of polymer-cli...?!?!?

**Comment by mercmobily** [comment]
Alright answering myself... It would be great to know if I got things right!

1) Polymer-cli no longer uses vulcanise; instead, it uses its internal logic to build things assuming that the app has a fixed entry point, a "shell" (that includes most needed modules) and fragments.

 2) I should make `my-main` the shell. It will get bundled so that it *includes* `my-app`  (or it should anyway)

3) I could use vulcanise, but... since Polymer-cli is there, and it works so awesomely, and since my app uses the pattern expected by polymer-cli, I should use that instead.

4) if I HAD TO use vulcanize for whatever reason, I will need to make sure to place a file that imports "everything" (including Polymer) and import *that one* in my fragments (although specific fragments might import extra modules, as long as they are not included in the "big bundle")

5) I really do have to use vulcanise, as long as https://github.com/Polymer/polymer-cli/issues/444 isn't fixed (and I realise this may take a while!) Is vulcanize going to be supported long term?

For the record, I was going insane trying to figure out why polymer-cli wouldn't work on my project when I finally found the problem and reported it: https://github.com/Polymer/polymer-cli/issues/444
